### PR TITLE
HPCC-25995 Default image version to match helm chart version

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [x] Cloud-compatibility
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by jakesmith — APPROVED** [review]
Looks sensible to me.
We could get the helm templates to check this version against the helm version and issue a warning, but a separate issue.

File: helm/hpcc/values.yaml [hunk 1]
```diff
@@ -5,9 +5,9 @@ global:
 
   image:
     ## It is recommended to name a specific version rather than latest, for any non-trivial deployment
-    ## For best results, the helm chart version and platform version should match. Leave this unset to use the
-    ## version that matches the version of the helm chart
-    #version: x.y.z 
+    ## For best results, the helm chart version and platform version should match. Do not override without
+    ## good reason as undefined behavior may result. 
+    version: {{ Chart.version }}
```
Comment (jakesmith) [inline helm/hpcc/values.yaml:10]: should this be {{ .Chart.version }} ?

File: helm/hpcc/templates/NOTES.txt [hunk 2]
```diff
@@ -11,7 +11,11 @@ This chart has defined the following HPCC components:
 {{-      if hasKey $names $component.name -}}
 {{-       required ( printf "Component name %s in section %s was already seen in section %s" $component.name $sectionName (get $names $component.name)) nil -}}
 {{-      else }}
-  {{ $sectionName }}.{{ $component.name }}
+  {{ $sectionName }}.{{ $component.name }}
```
Comment (jakesmith) [inline helm/hpcc/templates/NOTES.txt:14]: think there's a trailing whitespace here

Comment (richardkchapman) [reply helm/hpcc/templates/NOTES.txt]: Fixed and repushed

**Review by jakesmith — APPROVED** [review]
Looks good

**Comment by richardkchapman** [comment]
@ghalliday Please review/merge

**Review by ghalliday — APPROVED** [review]

**Pull request merged**